### DISCUSSION TITLE
‘获取m3u8 key...’ 之后闪退

### DISCUSSION THREAD

**Discussion by HBeing3** [nilaoda/N_m3u8DL-CLI#409] [Q&A]
N_m3u8DL-CLI v2.9.6 20210304...
Speed: waiting
Progress: waiting

                                                                                                                        13:37:23.884 文件名称：585fc98b71904b23a2bb3cd380edbdf9_4231311214_480p_enc_20210315133723
13:37:23.884 存储路径：E:\Download\ffmpeg-N-101538-g63344337f9-win64-gpl-shared-vulkan\bin\Downloads
13:37:23.896 开始解析
13:37:23.933 获取m3u8内容
13:37:24.615 解析m3u8内容
13:37:24.618 获取m3u8 key...

之后就退了。无法到写出meta.json和之后步骤
请问知道是什么原因吗？谢谢

**Comment by nilaoda** [comment]
获取不了key，是什么网站的视频？

**Reply by HBeing3** [reply]
是云学堂的视频，我是登录账号后到的这个地址，不知道你那用这个好不好使
https://streamows.yunxuetang.com/orgs/zkmba/knowledge/courseware/202003/585fc98b71904b23a2bb3cd380edbdf9_4231311214_480p_enc.m3u8

**Reply by nilaoda** [reply]
你这个网站返回的并不是直接的key，需要二次处理. 等下载器的下个版本支持吧

**Reply by HBeing3** [reply]
啊 好的，谢谢。

**Reply by HBeing3** [reply]
后来我研究了一下。貌似问题出在，URI 链接无法获得key，返回的是 {"code":"BAD_REQUEST","message":"lack of token in query string"}。
搜了半天，不知道怎么解决了。。。 唯一搜到了这个https://blog.csdn.net/liujiayu2/article/details/86083400，但他最后没能解密

**Reply by nilaoda** [reply]
> 啊 好的，谢谢。

这网站token也需要获取，m3u8里面没有带。所以不打算支持了，下载器不参与解析功能……
